TST: DeprecationWarning: SelectableGroups dict interface is deprecated. Use select.

Example log: https://github.com/astropy/astropy/runs/2321021984?check_suite_focus=true

Snippet (one of many): 

```
___________ ERROR collecting astropy/modeling/tests/data/__init__.py ___________
tests/lib/python3.7/site-packages/pytest_doctestplus/plugin.py:206: in collect
    module = self.fspath.pyimport()
tests/lib/python3.7/site-packages/py/_path/local.py:704: in pyimport
    __import__(modname)
astropy/modeling/__init__.py:10: in <module>
    from . import fitting
astropy/modeling/fitting.py:1764: in <module>
    populate_entry_points(entry_points().get('astropy.modeling', []))
tests/lib/python3.7/site-packages/importlib_metadata/__init__.py:312: in get
    flake8_bypass(self._warn)()
E   DeprecationWarning: SelectableGroups dict interface is deprecated. Use select.
```

--- a/astropy/io/misc/asdf/tests/__init__.py
+++ b/astropy/io/misc/asdf/tests/__init__.py
@@ -11,10 +11,15 @@ except ImportError:
 
 import pytest
 
-ep = [entry.name for entry in entry_points().get('asdf_extensions', [])]
+# TODO: Exclusively use select when Python minversion is 3.10
+eps = entry_points()
+if hasattr(eps, 'select'):
+    ep = [entry.name for entry in eps.select('asdf_extensions', [])]
+else:
+    ep = [entry.name for entry in eps.get('asdf_extensions', [])]
 ASDF_ENTRY_INSTALLED = 'astropy' in ep and 'astropy-asdf' in ep
 
-del entry_points, ep
+del entry_points, eps, ep
 
 if not ASDF_ENTRY_INSTALLED:
     pytest.skip('The astropy asdf entry points are not installed',

--- a/astropy/io/misc/asdf/tests/__init__.py
+++ b/astropy/io/misc/asdf/tests/__init__.py
@@ -14,7 +14,7 @@ import pytest
 # TODO: Exclusively use select when Python minversion is 3.10
 eps = entry_points()
 if hasattr(eps, 'select'):
-    ep = [entry.name for entry in eps.select('asdf_extensions', [])]
+    ep = [entry.name for entry in eps.select(group='asdf_extensions')]
 else:
     ep = [entry.name for entry in eps.get('asdf_extensions', [])]
 ASDF_ENTRY_INSTALLED = 'astropy' in ep and 'astropy-asdf' in ep

--- a/astropy/modeling/fitting.py
+++ b/astropy/modeling/fitting.py
@@ -1765,7 +1765,7 @@ def _populate_ep():
     # TODO: Exclusively use select when Python minversion is 3.10
     ep = entry_points()
     if hasattr(ep, 'select'):
-        populate_entry_points(ep.select('astropy.modeling', []))
+        populate_entry_points(ep.select(group='astropy.modeling'))
     else:
         populate_entry_points(ep.get('astropy.modeling', []))
 

--- a/astropy/modeling/fitting.py
+++ b/astropy/modeling/fitting.py
@@ -1761,4 +1761,13 @@ def populate_entry_points(entry_points):
                         'astropy.modeling.Fitter' .format(name)))
 
 
-populate_entry_points(entry_points().get('astropy.modeling', []))
+def _populate_ep():
+    # TODO: Exclusively use select when Python minversion is 3.10
+    ep = entry_points()
+    if hasattr(ep, 'select'):
+        populate_entry_points(ep.select('astropy.modeling', []))
+    else:
+        populate_entry_points(ep.get('astropy.modeling', []))
+
+
+_populate_ep()

